### DISCUSSION TITLE
How to set a Text Report Parameter default from a separate data set?

### DISCUSSION THREAD

**Discussion by pkav2000** [eclipse-birt/birt#1144] [Q&A]
Hi, I've a new Test report parameter which needs to have a default value set. The value needs to come from a reference table as it'll be set in a different part of the system.
Can anyone advise how to do this or point me at an idiots guide? Thanks.

**Comment by speckyspooky** [comment]
Hi, pkav2000,
BIRT has a method on the parameter to set the default value - "getDefultValueList".
There is the option given to set the default value.

Your challenge is that you have an (dynamic) data set. The problem is that BIRT haven't an UI-window to configure it.
But you are able to use your data set via API-Interface-calls.
I created an example for your based on the communitiy entry [https://www.eclipse.org/forums/index.php/t/361227/](url).
And you can take a look and change it to your case.

I attache my test example and my screen of my BIRT viewer. I created the demo with BIRT 4.8 and Java 11.


**1. BIRT demo report**
[default_parameter_with_data_engine.zip](https://github.com/eclipse/birt/files/10296751/default_parameter_with_data_engine.zip)


**2. BIRT viewer screen**
![2022-12-23 19_06_46-BIRT-viewer-default-parameter](https://user-images.githubusercontent.com/41593722/209389876-7fbc5d72-84ee-445f-a457-a3a6e950a04f.png)

**Reply by pkav2000** [reply]
This is brilliant! Thank you very much
